### PR TITLE
Update grandeur

### PR DESCRIPTION
**Pull request by erinyoung**
This is for Grandeur version 3.0.20230115 for ICA

This subdirectory is sync'd with my local copy. I'm hoping to make changes to the original repo in order to make updates easier. 

I am expecting updates semi-frequently - at least every time refseq updates.

### CODE REVIEW

**Comment by erinyoung** [comment]
We no longer use this version, so I'm closing this PR

**Comment by erinyoung** [comment]
WAIT!!!! JUST KIDDING!!! I got confused with a different PR.

**Comment by erinyoung** [comment]
I still need to change the database paths and xml file. I'm hoping to get to that sometime early next week.

**Comment by erinyoung** [comment]
@jwarnn , it's ready for some additional review

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+                <pd:parameter code="outdir" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>outdir</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>out/grandeur</pd:value>
+                </pd:parameter>
+                <pd:parameter code="minimum_reads" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>minimum_reads</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ``` <pd:stringType/>``` to ```<pd:integerType/>```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                <pd:parameter code="msa" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>msa</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>false</pd:value>
+                </pd:parameter>
+                <pd:parameter code="roary_min_genes" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>roary_min_genes</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: <pd:stringType/> to <pd:integerType/>

File: Grandeur-3.0.20230120/modules/amrfinderplus.nf [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  tag           "${sample}"
+  label         "medcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/ncbi-amrfinderplus:3.10.36'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  cpus 3
+  memory 1.GB
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-medium'
+
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/amrfinderplus.nf:11]: For whatever reason this process is slow on ICA after a few dozen samples. I would change to 
``` 
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
  cpus          7
  memory        26.GB
```

File: Grandeur-3.0.20230120/modules/blast.nf [hunk 4]
```diff
@@ -0,0 +1,39 @@
+process blastn {
+  tag           "${sample}"
+  label         "medcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/blast:2.13.0'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
+  cpus 7
+  memory 25.GB
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/blast.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-xlarge'
  cpus          14
  memory        60.GB
```

File: Grandeur-3.0.20230120/modules/fastani.nf [hunk 5]
```diff
@@ -0,0 +1,51 @@
+process fastani {
+  tag           "${sample}"
+  label         "medcpus"
+  stageInMode   "copy"
+  publishDir    path: params.outdir, mode: 'copy', pattern: 'logs/*/*log'
+  publishDir    path: params.outdir, mode: 'copy', pattern: 'fastani/*' 
+  container     'staphb/fastani:1.33'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-medium'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/fastani.nf]:
```
pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
  cpus          7
  memory        26.GB
```

File: Grandeur-3.0.20230120/modules/kraken2.nf [hunk 6]
```diff
@@ -0,0 +1,89 @@
+process kraken2_fastq {
+  tag           "${sample}"
+  label         "maxcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/kraken2:2.1.2-no-db'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/kraken2.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-xlarge'
  cpus          14
  memory        60.GB
```

File: Grandeur-3.0.20230120/modules/mash.nf [hunk 7]
```diff
@@ -0,0 +1,143 @@
+process mash {
+  tag           "${sample}"
+  label         "medcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/mash:2.3'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/mash.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-xlarge'
  cpus          14
  memory        60.GB
```

File: Grandeur-3.0.20230120/modules/shigatyper.nf [hunk 8]
```diff
@@ -0,0 +1,51 @@
+process shigatyper {
+  tag           "${sample}"
+  label         "medcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/shigatyper:2.0.1'
+  stageInMode   'copy'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-medium'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/shigatyper.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
  cpus          7
  memory        26.GB
```

File: Grandeur-3.0.20230120/modules/spades.nf [hunk 9]
```diff
@@ -0,0 +1,41 @@
+process spades {
+  tag           "${sample}"
+  label         "maxcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/spades:3.15.5'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/spades.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-xlarge'
  cpus          14
  memory        60.GB
```

File: Grandeur-3.0.20230120/modules/iqtree2.nf [hunk 10]
```diff
@@ -0,0 +1,42 @@
+process iqtree2 {
+  tag           "Pylogenetic Analysis"
+  label         "maxcpus"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/iqtree2:2.1.2'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/iqtree2.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-xlarge'
  cpus          14
  memory        60.GB
```

File: Grandeur-3.0.20230120/modules/fastqscan.nf [hunk 11]
```diff
@@ -0,0 +1,42 @@
+process fastqscan {
+  tag           "${sample}"
+  stageInMode   "copy"
+  publishDir    params.outdir, mode: 'copy'
+  container     'staphb/fastq-scan:1.0.1'
+  maxForks      10
+  errorStrategy { task.attempt < 2 ? 'retry' : 'ignore'}
+  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-medium'
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/modules/fastqscan.nf]:
```
  pod annotation: 'scheduler.illumina.com/presetSize', value: 'standard-large'
  cpus          7
  memory        26.GB
```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:value>blast_db</pd:value>
+                </pd:parameter>
+                <pd:parameter code="mash_db" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>mash_db</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>new_mash</pd:value>
+                </pd:parameter>
+                <pd:parameter code="sra_accessions" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="sra_accessions" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value>true</pd:value>
+                </pd:parameter>
+                <pd:parameter code="datasets_max_genomes" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>datasets_max_genomes</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>5</pd:value>
+                </pd:parameter>
+                <pd:parameter code="iqtree2_outgroup" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="iqtree2_outgroup" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 14]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="bbduk_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>bbduk_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>k=31 hdist=1</pd:value>
+                </pd:parameter>
+                <pd:parameter code="bbmap_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="bbmap_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 15]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="blastn_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>blastn_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>-max_target_seqs 10 -max_hsps 1 -evalue 1e-25</pd:value>
+                </pd:parameter>
+                <pd:parameter code="blobtools_create_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="blobtools_create_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>blobtools_create_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="blobtools_view_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="blobtools_view_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="blobtools_plot_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>blobtools_plot_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>--format png -r species</pd:value>
+                </pd:parameter>
+                <pd:parameter code="blobtools_bbmap_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="blobtools_bbmap_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 18]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="fastani_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>fastani_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>--matrix</pd:value>
+                </pd:parameter>
+                <pd:parameter code="fasterqdump_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="fasterqdump_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 19]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="fastp_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>fastp_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>--detect_adapter_for_pe</pd:value>
+                </pd:parameter>
+                <pd:parameter code="fastqc_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="fastqc_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 20]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>fastqc_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="fastqscan_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="fastqscan_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 21]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value>-t RANDOM -m GTR+F+I -bb 1000 -alrt 1000</pd:value>
+                </pd:parameter>
+                <pd:parameter code="kleborate_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>kleborate_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>-all</pd:value>
+                </pd:parameter>
+                <pd:parameter code="kraken2_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="kraken2_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 22]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value>-m 2</pd:value>
+                </pd:parameter>
+                <pd:parameter code="mash_dist_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>mash_dist_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>-v 0 -d 0.5</pd:value>
+                </pd:parameter>
+                <pd:parameter code="mlst_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="mlst_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 23]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>mlst_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="multiqc_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="multiqc_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 24]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>multiqc_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="plasmidfinder_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="plasmidfinder_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 25]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="prokka_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>prokka_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>--mincontiglen 500 --compliant --locustag locus_tag --centre STAPHB</pd:value>
+                </pd:parameter>
+                <pd:parameter code="quast_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="quast_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 26]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>quast_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="roary_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="roary_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 27]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="seqsero2_options" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>seqsero2_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>-m a -b mem</pd:value>
+                </pd:parameter>
+                <pd:parameter code="serotypefinder_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="serotypefinder_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 28]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>serotypefinder_options</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value/>
+                </pd:parameter>
+                <pd:parameter code="shigatyper_options" minValues="1" maxValues="1" classification="USER">
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ```<pd:parameter code="shigatyper_options" minValues="0" maxValues="1" classification="USER">```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 29]
```diff
@@ -0,0 +1,295 @@
...
+                    <pd:label>minimum_reads</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>10000</pd:value>
+                </pd:parameter>
+                <pd:parameter code="kraken2_db" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>kraken2_db</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>kraken2_db</pd:value>
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ``` <pd:value>./kraken2_db</pd:value> ```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 30]
```diff
@@ -0,0 +1,295 @@
...
+                </pd:parameter>
+                <pd:parameter code="blast_db" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>blast_db</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>blast_db</pd:value>
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ``` <pd:value>./blast_db</pd:value> ```

File: Grandeur-3.0.20230120/XML_Configuration.xml [hunk 31]
```diff
@@ -0,0 +1,295 @@
...
+                </pd:parameter>
+                <pd:parameter code="mash_db" minValues="1" maxValues="1" classification="USER">
+                    <pd:label>mash_db</pd:label>
+                    <pd:description locked="false"></pd:description>
+                    <pd:stringType/>
+                    <pd:value>new_mash</pd:value>
```
Comment (jwarnn) [inline Grandeur-3.0.20230120/XML_Configuration.xml]: ``` <pd:value>./new_mash</pd:value> ```

**Review by jwarnn — CHANGES_REQUESTED** [review]
OK. A number of changes to XML file. A few changes to the modules to give them a little more juice. With these changes we should be good to go.

**Comment by erinyoung** [comment]
I think I made all the required changes!

**Review by jwarnn — APPROVED** [review]

**Pull request merged**